services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -1312,8 +1312,11 @@ private Integer fetchSystemStoreSchemaId(String clusterName, String storeName, S
       }
       return valueSchemaId;
     }
-    ControllerClient controllerClient = ControllerClient
-        .constructClusterControllerClient(clusterName, getLeaderController(clusterName).getUrl(false), sslFactory);
+
+    ControllerClient controllerClient = ControllerClient.constructClusterControllerClient(
+        clusterName,
+        getLeaderController(clusterName).getUrl(getSslFactory().isPresent()),
+        sslFactory);
```

Comment:
I wonder whether we need to change other places for `getUrl`. For example, in `com.linkedin.venice.controllerapi.ControllerClient#discoverLeaderController`, we still use `getUrl` instead of `getSecureUrl`. 

Do we plan to clean up all getUrl() or just getUrl() for Instance?

Reply:
I think we only want to change getUrl of Instance.
You pointed to the "url" in LeaderControllerResponse, which is actually used like unsecuredUrl. LeaderControllerResponse also has secureUrl